robottelo/hosts.py

```diff
@@ -222,7 +222,12 @@ def teardown(self):
                 self.nailgun_host.delete()
             self.unregister()
         # Strip most unnecessary attributes from our instance for checkin
-        keep_keys = set(self.to_dict()) | {'release', '_prov_inst', '_cont_inst'}
+        keep_keys = set(self.to_dict()) | {
+            'release',
+            '_prov_inst',
+            '_cont_inst',
+            '_skip_context_checkin',
```

Comment:
we technically don't need this anymore since Broker now uses threaded checkins, but I won't block on removing it.

Reply:
I believe we can safely remove this section altogether. It appears to serve a different purpose than the rest of the code so it might be better suited for a separate pull request.